src/api06/changeset_upload_handler.cpp

```diff
@@ -68,6 +68,26 @@ changeset_upload_responder::changeset_upload_responder(mime::type mt,
     }
   }
 
+  if (global_settings::get_bbox_size_limiter_upload()) {
+
+    auto const cs_bbox = handler.get_bbox();
+
+    if (!(cs_bbox == bbox_t()))   // valid bbox?
+    {
+      auto const max_bbox_size = upd.get_bbox_size_limit(*user_id);
+
+      if (cs_bbox.linear_size() > max_bbox_size) {
```

Comment:
This is only checking the bbox of the current upload. 
https://github.com/openstreetmap/openstreetmap-website/pull/4908/files#diff-33e24e983f7ae5829e0e919c3843ccaaa05be35610607ec2fd486636d2c14e56R131-R133 checks the full bounding box of the changeset after expansion.

The check needs to be moved down after `changeset_updater->update_changeset(new_changes, handler.get_bbox());` and use the expanded bounding box `changeset_updater.cs_bbox` (which is currently not accessible from outside, so a getter method is needed, or the expanded bbox should be returned from `update_changeset()`)

Reply:
Good catch, I remember now why the previous implementation (that is no longer available) performed the check in ApiDB_Changeset_Updater::update_changeset. I think I'll move it there now.

Reply:
So I tested the Rails implementation with multiple uploads. The second one would fail b/c the resulting bbox is too large. I think I also need to add this scenario as a test case.

Reply:
This should be fixed now, I added a getter-method + added a test case.